connector/src/test/scala/com/microsoft/kusto/spark/utils/ContainerProviderTest.scala

```diff
@@ -75,6 +80,57 @@ class ContainerProviderTest extends AnyFlatSpec with Matchers with MockFactory {
     the[RuntimeException] thrownBy containerProvider.getContainer should have message "Failed to allocate temporary storage"
   }
 
+  "ContainerProvider" should "retry and return a container in case of a temporary HTTPException" in {
+    val extendedMockClient = mock[ExtendedKustoClient]
+    val extendedMockNoRootExceptionFail = mock[ExtendedKustoClient]
+    val extendedMockClientIOExceptionFail = mock[ExtendedKustoClient]
+    val kustoOperationResult = new KustoOperationResult(readTestSource("storage-result.json"), "v1")
+    val clusterAlias = "ingest-cluster"
+    val command = ".get ingestion resources"
+    val ingestProviderEntryCreator = (c: ContainerAndSas) => c
+    /*
+      Invoke and test
+     */
+    val containerProvider = new ContainerProvider(extendedMockClient, clusterAlias, command,
+      ingestProviderEntryCreator, CACHE_EXPIRY_SEC)
+
+    extendedMockClient.executeDM _ expects(command, None, *) throws new DataServiceException(clusterAlias,
+      "IOError when trying to retrieve CloudInfo",(new HttpHostConnectException(
+      new IOException("Connection timed out"),
```

Comment:
I think the error is built 
new HttpHostConnectException(new java.net.ConnectException("..